src/gridfs/options.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    concern::{ReadConcern, WriteConcern},
+    selection_criteria::SelectionCriteria,
+};
+use serde::Deserialize;
+use typed_builder::TypedBuilder;
+
+use bson::Document;
+
+/// Contains the options for creating a [`GridFsBucket`].
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFsBucketOptions {
+    /// The bucket name. Defaults to 'fs'.
+    pub bucket_name: Option<String>,
+
+    /// The chunk size in bytes used to break the user file into chunks. Defaults to 255 KiB.
+    pub chunk_size_bytes: Option<i32>,
+
+    /// The write concern. Defaults to the write concern of the database.
+    pub write_concern: Option<WriteConcern>,
+
+    /// The read concern. Defaults to the read concern of the database.
+    pub read_concern: Option<ReadConcern>,
+
+    /// The read preference. Defaults to the read preference of the database.
+    pub read_preference: Option<SelectionCriteria>,
+}
+
+/// Contains the options for creating a [`GridFsStream`] to upload a file to a
+/// [`GridFsBucket`].
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFsUploadOptions {
+    /// The number of bytes per chunk of this file. Defaults to the `chunk_size_bytes` specified
+    /// in the [`GridFsBucketOptions`].
+    pub chunk_size_bytes: Option<i32>,
+
+    /// User data for the 'metadata' field of the files collection document.
+    pub metadata: Option<Document>,
+}
+
+/// Contains the options for creating [`GridFsStream`] to retrieve a stored file
+/// from a [`GridFsBucket`].
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFsDownloadByNameOptions {
+    /// Which revision (documents with the same filename and different `upload_date`)
+    /// of the file to retrieve. Defaults to -1 (the most recent revision).
+    ///
+    /// Revision numbers are defined as follows:
+    /// 0 = the original stored file
+    /// 1 = the first revision
+    /// 2 = the second revision
+    /// etc...
+    /// -2 = the second most recent revision
+    /// -1 = the most recent revision
+    pub revision: Option<i32>,
```

Comment:
it might be a bit more user-friendly to model this with an enum since there are a set of known values corresponding to constants

Reply:
I'm not sure I understand what you mean. Say if someone wanted to get the eighth revision for a file, would we create enum variants for every single revision?
e.g.
```
enum FileRevision {
    OriginalStoredFile,
    First,
    Second,
    // ... and so on
    SecondMostRecent,
    MostRecent,
}

Reply:
per our conversation offline I misunderstood the possible values of this enum and didn't realize it could be any number, not just one of a few, so I retract this suggestion